components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1974,11 +1975,12 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
 
         const user = await this.checkAndBlockUser("getSnapshots");
 
-        const workspace = await this.workspaceDb.trace(ctx).findById(workspaceId);
-        if (!workspace || workspace.ownerId !== user.id) {
+        // we use the workspacService which checks if the requesting user has access to the workspace. If that is the case they have access to snapshots as well.
+        // below is the old permission check which would also check if the user has access to the snapshot itself. This is not the case anymore.
+        const workspace = await this.workspaceService.getWorkspace(user.id, workspaceId);
```

Comment:
What should I test here exactly?

Reply:
Create a workspace from a snapshot context, to validate the fact that all users (incl member + strangers) can create workspace from a Snapshot.

Reply:
Since you know how painful it would be to write a test against gitpod-server-iml. Do you suggest to extract this logic into a service and write a test?